### PR TITLE
[feature] implement upload targets

### PR DESCRIPTION
**Pull request by trombik**
this PR implements "reallyenglish:upload"

also:

* refactor namespaces and remove `args`
* add a simple test to see boxes boot with a single CPU

the most relevant targets for development are:

```
rake reallyenglish:test:all                            # build and test all boxes
rake reallyenglish:test:centos-7.3-x86_64              # Test centos-7.3-x86_64
rake reallyenglish:test:debian-8.7-amd64               # Test debian-8.7-amd64
rake reallyenglish:test:debian-8.8-amd64               # Test debian-8.8-amd64
rake reallyenglish:test:debian-9.0-amd64               # Test debian-9.0-amd64
rake reallyenglish:test:freebsd-10.3-amd64             # Test freebsd-10.3-amd64
rake reallyenglish:test:openbsd-5.9-amd64              # Test openbsd-5.9-amd64
rake reallyenglish:test:openbsd-6.0-amd64              # Test openbsd-6.0-amd64
rake reallyenglish:test:openbsd-6.1-amd64              # Test openbsd-6.1-amd64
rake reallyenglish:test:openbsd-6.2-amd64              # Test openbsd-6.2-amd64
rake reallyenglish:test:ubuntu-14.04-amd64             # Test ubuntu-14.04-amd64
rake reallyenglish:test:ubuntu-16.04-amd64             # Test ubuntu-16.04-amd64
```

### CODE REVIEW

File: Rakefile [hunk 1]
```diff
@@ -86,6 +86,8 @@ end
 
 namespace :reallyenglish do
   require 'yaml'
+  require 'vagrant_cloud'
+  vagrantcloud_user_name = ENV["VAGRANTCLOUD_USERNAME"] || "trombik"
```
Comment (mitsururike) [inline Rakefile:90]: i do not understand why `VAGRANTCLOUD_USERNAME` is not mandatory and  the `vagrantcloud_user_name` has the default value.

Comment (trombik) [reply Rakefile:90]: https://github.com/reallyenglish/packer-templates/pull/56/commits/2aef74a579811fcfc1481c3d215fdcec11069646 clarifies the reason.

**Review by mitsururike — COMMENTED** [review]
question: what happens if the same version of the box already exists in the vagrant cloud? is it overwritten by the new one or uploading is blocked?

**Comment by trombik** [comment]
the target overwrites the box.

**Comment by mitsururike** [comment]
if the box has been released, it must not be overwritten without bumping version number.
if it is difficult to implement to block overwriting the released box, the procedure of
uploading which includes checking version numbers should be written.

**Comment by trombik** [comment]
that is what we have been doing, and i am not trying to fix it in this PR. creating an issue is preferred.

**Review by mitsururike — APPROVED** [review]

**Pull request merged**